cli/dataset.go

```diff
@@ -194,6 +195,14 @@ func (c *viamClient) downloadDataset(dst, datasetID string, includeJSONLines boo
 			}
 		}()
 	}
+	resp, err := c.datasetClient.ListDatasetsByIDs(context.Background(),
+		&datasetpb.ListDatasetsByIDsRequest{Ids: []string{datasetID}})
+	if err != nil {
+		return errors.Wrapf(err, "error getting dataset ID")
+	}
+	if len(resp.GetDatasets()) == 0 {
```

Comment:
Included this check because if the id can be converted to a valid ObjectID but there is no datasets returned, then the ListDatasetsByID should not return an error, but we still need to return an error from downloadDataset.